week3/lists.js

```diff
@@ -0,0 +1,98 @@
+const express = require('express');
+const router = express.Router();
+const db = require('./db');
+
+/* Get lists of logged in user */
+router.get('/', async (req, res) => {
+  const user = req.session.user;
+  if (user) {
+    const lists = await db.query('select * from todo_lists where user_id = ?', [user.id]);
+    res.json(lists);
+  }
+  else { res.status(401).end('Unauthorized'); }
+})
+
+/* Get list (by its id) */
+router.get('/:id', async (req, res) => {
+  const user = req.session.user;
+  if (user) {
+    const lists = await db.query('select * from todo_lists where id = ?', [req.params.id]);
+    const list = lists[0];
+    if (list.user_id !== user.id) {
+      res.status(401).end('Unauthorized');
+    }
+
+    list.items = await db.query('select * from todo_items where list_id = ? ', [list.id]);
+    res.json(list);
+  }
+  else {
+    res.status(401).end('Unauthorized');
+  }
+})
+
+/* Create a new list */
+router.post('/', async (req, res) => {
+  const user = req.session.user;
+  console.log(user);
+  if (user) {
+    const list_title = req.body.title;
+    if (!list_title) {
```

Comment:
Nice validation. But maybe also expect a length of something like 2 or more chars? =)